controllers/managedocs_controller.go

```diff
@@ -825,9 +825,11 @@ func (r *ManagedOCSReconciler) reconcileOnboardinValidationSecret() error {
 		if err := r.own(r.onboardingValidationKeySecret); err != nil {
 			return err
 		}
+		onboardingValidationData := "-----BEGIN PUBLIC KEY-----\n" + r.addonParams[onboardingValidationKey] + "\n-----END PUBLIC KEY-----"
```

Comment:
A couple of points
1. use `fmt` instead of manual string concatenation 
2. trim (for whitespaces) the passed `r.addonParams[onboardingValidationKey]` value